ARROW-17731: [Website] Add Flight SQL JDBC blog post

lidavidm opened the pull request:
I intend to pair this with ARROW-17732, but haven't yet written that up

lidavidm commented:
Rendered

![image](https://user-images.githubusercontent.com/327919/190271603-c1fec5bf-1f51-40fb-9c33-7be175c6f286.png)

lidavidm commented:
See https://github.com/apache/arrow/pull/14137

lidavidm commented:
Added a reference to the ODBC driver as well.

rapinderj33 commented:
Hi David, for the second paragraph under "Why JDBC", I think talking about ADBC takes away from the JDBC discussion and may confuse the user. 

Also whenever you mention Dremio can you add this link to it: https://www.dremio.com/resources/guides/apache-arrow/. 

Also after saying after courtesy of the grant by Dremio.... Can the next sentence include our tag line: "Dremio, the easy and open data lakehouse platform. "

jasonhughes248 commented on _drafts/arrow-flight-sql-jdbc.md:
```diff
@@ -0,0 +1,112 @@
+and Flight SQL, and increases the reach of Arrow-based applications.
+Additionally, an [ODBC driver implementation][dremio-odbc] based on
+Flight SQL is also available, though it is not part of the Arrow
+project.
```
can we make that last sentence "though it is not part of the Arrow project yet due to dependency licensing issues."

jasonhughes248 commented:
Hi David, made a couple comments. looks good!

lidavidm commented:
<details>
<summary>Rendered</summary>

![image](https://user-images.githubusercontent.com/327919/196201382-92066575-c891-408d-acd8-5920bb52bcbc.png)

</details>

Thanks for the feedback. Changes:

- Reorganized material to flow better, and emphasize the unique advantages of Flight SQL
- Consolidated "Background" into the intro
- Removed broken link since docs PR is not yet reviewed
- I want to be careful with marketing taglines as this post is meant to be about the Arrow ecosystem, but I added a tagline describing Dremio as a data lakehouse platform. @rapinderj33 is that ok?
- And I'm not sure if you wanted to link every mention of Dremio to the Arrow guide specifically, but I put a reference to that alongside the rest of the documentation.
- Changed ADBC to be a minor reference
- Link to Jason's presentation
- Mention the current licensing issues for the ODBC driver

lidavidm commented:
Latest version rendered: https://amazing-cocada-24b100.netlify.app/blog/2022/10/19/arrow-flight-sql-jdbc/

nealrichardson commented on _posts/2022-10-19-arrow-flight-sql-jdbc.md:
```diff
@@ -0,0 +1,115 @@
+database can support JDBC users without implementing additional APIs
+or its own JDBC driver.
+
+[^1]: Despite the name, Flight SQL is **not** a SQL dialect, nor is it
```
This seems important and worth reiterating--maybe put it in the paragraph itself instead of a footnote/

nealrichardson commented on _posts/2022-10-19-arrow-flight-sql-jdbc.md:
```diff
@@ -0,0 +1,115 @@
+
+The JDBC driver was merged for the Arrow 10.0.0 release, and the
+[source code][impl] can be found in the Arrow repository.  Once Arrow
+10.0.0 is released (which is planned for October/November 2022),
```
Revise this now that the RC is happening? And/or consider holding to publish with the release announcement--"you can check it out now" is a better call-to-action than "someday soon it will be released"

nealrichardson reviewed (approved):
Some minor notes but otherwise LGTM

lidavidm commented:
@rapinderj33 we'll push this once 10.0.0 officially releases, and I'll patch in download links and fix the date/time on the post before that. The vote is still underway, so hopefully that will be soon.

lidavidm commented:
Will go out 2022-11-01
